ground/src/main/java/com/google/android/ground/ui/datacollection/tasks/AbstractTaskFragment.kt

```diff
@@ -197,6 +219,31 @@ abstract class AbstractTaskFragment<T : AbstractTaskViewModel> : AbstractFragmen
 
   @TestOnly fun getButtonsIndex() = buttonsIndex
 
+  private fun showLoiNameDialog(initialTextValue: String) {
+    (view as ViewGroup).addView(
+      ComposeView(requireContext()).apply {
+        setContent {
+          val openAlertDialog = remember { mutableStateOf(Pair(initialTextValue, true)) }
+          when {
```

Comment:
The state-Hoisting approach can be very helpful, if, in future checks or logic increases, we have to move from Pair to other options, which will require more testing and more time. 

Also, currently, it's not much readable, as from `true`, it's hard to understand what it refers to.

Reply:
^I will make the Pair fields a little more clear, let me push that right now.